R/model.R

```diff
@@ -1285,7 +1311,12 @@ generate_quantities <- function(fitted_params,
            call. = FALSE)
     }
   }
-
+  if (is.null(self$cpp_options()[["stan_opencl"]])
+      && !is.null(opencl_ids)) {
+     stop("'opencl_ids' is set but the model was not compiled with for use with OpenCL.",
+           "\nRecompile the model with the 'cpp_options = list(stan_opencl = TRUE)'",
+           call. = FALSE)   
+  }
```

Comment:
Since this code is duplicated 4 times (sample, optimize, variational, gq), does it make sense to have a small convenience function to do this check?

Reply:
Good call! Thanks!